test/request_test.rb

```diff
@@ -0,0 +1,91 @@
+require File.expand_path(File.join(__FILE__, '../test_helper'))
+
+class RequestTest < Minitest::Test
+  # Unauthenticated requests should return an AuthenticationError.
+  def test_unauthenticated_request
+    without_authentication
+
+    raised = assert_raises BlockScore::NoAPIKeyError do
+      create_person
+    end
+
+    assert_equal 'No API key was provided.', raised.message
+
+    # reset API key or all tests run afterwards will fail.
+    with_authentication
+  end
+
+  def test_not_found_error
+    with_authentication
+
+    raised = assert_raises BlockScore::NotFoundError do
+      BlockScore::Person.retrieve('404')
+    end
+
+    assert_equal 'not_found_error', raised.error_type
+    assert_equal 404, raised.http_status
+
+    msg = '(Type: not_found_error) Person with ID ab973197319713ba could not be found (Status: 404)'
+    assert_equal msg, raised.message
+  end
+
+  def test_invalid_request_error
+    raised = assert_raises BlockScore::InvalidRequestError do
+      BlockScore::Person.retrieve('400')
+    end
+
+    assert_equal 'invalid_request_error', raised.error_type
+    assert_equal 400, raised.http_status
+
+    msg = '(Type: invalid_request_error) One of more parameters is invalid. (name_first) (Status: 400)'
+    assert_equal msg, raised.message
+  end
+
+  def test_api_error
+    with_authentication
+
+    raised = assert_raises BlockScore::APIError do
+      BlockScore::Person.retrieve('500')
+    end
+
+    assert_equal 'api_error', raised.error_type
+    assert_equal 500, raised.http_status
+
+    msg = '(Type: api_error) An error occurred. (Status: 500)'
+    assert_equal msg, raised.message
+  end
+
+  def test_authentication_error
+    raised = assert_raises BlockScore::AuthenticationError do
+      BlockScore::Person.retrieve('401')
+    end
+
+    assert_equal 'authentication_error', raised.error_type
+    assert_equal 401, raised.http_status
+
+    msg = '(Type: authentication_error) The provided API key is invalid. (Status: 401)'
+    assert_equal msg, raised.message
+  end
+
+  def test_socket_error
+    stub_request(:get, /.*api\.blockscore\.com\/people\/socket_error/).
+      to_raise(SocketError)
+
+    raised = assert_raises(BlockScore::APIConnectionError) do
+      BlockScore::Person.retrieve('socket_error')
+    end
+
+    assert_equal 'Exception from WebMock', raised.message
+  end
+
+  def test_connection_refused
+    stub_request(:get, /.*api\.blockscore\.com\/people\/connection_refused/).
```

Comment:
Style/DotPosition: Place the . on the next line, together with the method name.<br>Style/RegexpLiteral: Use %r for regular expressions matching more than 1 '/' character.